canonical-web-and-design/maas-ui
# Support ctrl clicking links

## Done

- Move all basenames to the shared workspace.
- Add utils for generating and navigation to new/legacy urls.
- Update links to use utils.
- Added a legacy link component to use in the ui workspace.
- Updated proxy to redirect to /MAAS/r.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check /MAAS/r/ redirects to /MAAS/r/machines.
- In the action menu on status columns check that "See logs" and "See events" navigate normally and with ctrl click.
- Can navigate normally and with ctrl click via the header to and from legacy and react clients.
- Can ctrl click (plus normal) on machines and subnets etc in the machine listA

## Fixes

Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/1441.
Fixes: https://github.com/canonical-web-and-design/maas-ui/issues/1265.
